src/reconstruction.jl

```diff
@@ -306,12 +359,13 @@ function estimate_delay(x::AbstractVector, method::String)
         τ = exponential_decay(c)
         return round(Int,τ)
     #Need a package that can be precompiled...
-    # elseif method=="mutual_inf"
-    #     m = get_mutual_information(s,s)
-    #     for i=1:length(x)÷10
-    #         n = get_mutual_information(s[1:end-i],s[1+i:end])
-    #         m > n && break
-    #     end
+    elseif method=="mutual_inf"
+        m = mutual_info(x,x; k=k)
+        for i=1:maxtau
+            n = mutual_info(x[1:end-i],x[1+i:end]; k=k)
```

Comment:
`x[1:end-i]` could be replaced by `(@view x[1:end-1])`. It makes not much of a difference, because the `mutual_info` is much slower, but on Julia 0.7 `views` will be super optimized so its nice to plan ahead :)